modules/RoutingContext.js

```diff
@@ -5,15 +6,32 @@ import getRouteParams from './getRouteParams'
 
 const { array, func, object } = React.PropTypes
 
-/**
- * A <RoutingContext> renders the component tree for a given router state
- * and sets the history object and the current location in context.
- */
 class RoutingContext extends Component {
 
   getChildContext() {
     const { history, location } = this.props
-    return { history, location }
+    const router = {
+      push(...args) {
+        history.push(...args)
+      },
+      replace(...args) {
+        history.replace(...args)
+      },
+      addRouteLeaveHook(...args) {
+        return history.listenBeforeLeavingRoute(...args)
+      },
+      isActive(...args) {
+        return history.isActive(...args)
+      }
```

Comment:
This can be rewritten to make that less messy:
```js
const router = {
      push: history.push,
      replace: history.replace,
      addRouteLeaveHook: history.listenBeforeLeavingRoute,
      isActive: history.isActive,
      createHref: history.createHref,
      go: history.go,
      goBack: history.goBack,
      goForward: history.goForward
    }```